rust/kernel/spi.rs

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use crate::bindings;
+use crate::c_types;
+use crate::error::{Error, Result};
+use crate::CStr;
+use alloc::boxed::Box;
+use core::pin::Pin;
+
+#[derive(Clone, Copy)]
+pub struct SpiDevice(*mut bindings::spi_device);
+
+impl SpiDevice {
+    pub fn from_ptr(dev: *mut bindings::spi_device) -> Self {
```

Comment:
This should be unsafe. Using an invalid pointer would allow a crash/UB inside the methods on `Spi`.

Reply:
Good catch, this is an oversight. Thanks :)